test/e2e/start_test.go

```diff
@@ -83,7 +82,6 @@ var _ = Describe("Podman start", func() {
 		if podmanTest.RemoteTest {
 			Skip("Container-start name check doesn't work on remote client. It always returns the full ID.")
 		}
-		Expect(session.OutputToString()).To(Equal(name))
```

Comment:
Why are we dropping this? I think this still seems like something we want to pass?

Reply:
we are printing the ID instead of the short name, now the output looks like:

```
podman start foo
98ea6e4f216f2fb4b69fff9b3a44842c38686ca685f3f55dc48c5d3fb1107be4
```

I can drop this patch but is it a requirement to print the short name?  How would we do it?  Doing the reverse mapping in the frontend?

Reply:
I think the key here is that we print the input that the user gave - so if the user typed `foo` we print `foo`, if they typed `98ea` we print `98ea`

Reply:
how would we handle multiple containers?

`podman start foo1 foo2 foo3`..

Do we have any guarantee the result keeps the same order?

Reply:
Hmmm. Back in v1, we had the loop converting shortname -> container in main, so we could easily look this up... @jwhonce @baude Thoughts here?

Reply:
this has plagued me a little while now ... i would prefer full id but i think docker does names? the problem is with the new approach we loose context of whether we were passed a name vs an ID so we also dont know how to respond?

Reply:
yes, Docker returns the name you've passed as argument and probably this is the reason why we have added this test.

Possible solutions if we want to keep the current behavior:
- in addition to the ID, pass back the name specified (quite ugly)
- enforce the returned list to keep the same order
- retrieve the ID from the client (do a lookup for each input)
- don't care about Docker, `create`, `run` and `stop` already print the full ID

I like the last two options more, we don't need to change the API and if the ID lookup turns out to be expensive we can drop it/optimize later.

@vrothberg any thought?

Reply:
I checked Docker 19.03.8 which always returns what the user specified:
* start foo -> foo
* start 123 -> foo
* start $fullID -> $fullID

As pointed out above, if multiple containers are specified we need to know which containers have been started: `$ podman start 1 2 3 doesNotExist 4` ... we need to know that `doesNotExist` could not be started.

While I am not 100 percent sure if we could extract that data already, I suggest to make things a bit more explicit and embed the user input in the return type. We could extend `ContainerStartReport` with `RawInput` (or sth similar): https://github.com/containers/libpod/blob/master/pkg/domain/entities/containers.go#L228

The output in `cmd/...` could then be done by iterating over the report slice:
```golang
errors := []
for _, r := range startReports {
 if r.Err != nil { // append to errors and continue }
 if r.RawInput == "" { // LATEST
   fmt.Println(r.ID)
 }  else 
   fmt.Println(r.RawInput)
  }
}
```